addons/account/tests/test_tax_report.py

```diff
@@ -178,3 +178,72 @@ def test_tax_report_change_country(self):
+            'amount_type': 'percent',
+            'amount': 25,
+            'country_id': self.tax_report_1.country_id.id,
+            'type_tax_use': 'sale',
+            'invoice_repartition_line_ids': [
+                (0, 0, {'factor_percent': 100, 'repartition_type': 'base'}),
+                (0, 0, {
+                    'factor_percent': 100,
+                    'repartition_type': 'tax',
+                    'tag_ids': [Command.link(test_tag.id)],
+                }),
+            ],
+            'refund_repartition_line_ids': [
+                (0, 0, {'factor_percent': 100, 'repartition_type': 'base'}),
+                (0, 0, {'factor_percent': 100, 'repartition_type': 'tax'}),
+            ],
+        })
+
+        # Make sure the fiscal country allows using this tax directly
+        self.env.company.account_fiscal_country_id = self.test_country_1
+
+        test_invoice = self.env['account.move'].create({
+            'move_type': 'out_invoice',
+            'partner_id': self.partner_a.id,
+            'date': '1992-12-22',
+            'invoice_line_ids': [
+                (0, 0, {'quantity': 1, 'price_unit': 42, 'tax_ids': [Command.set([test_tax.id])]}),
+            ],
+        })
+        test_invoice.action_post()
+
+        tax_report_line.unlink()
+        plus_tags_after = self._get_tax_tags(self.test_country_1, tag_name=tag_name, active_test=False).filtered(lambda tag: tag.name[:1] == '+')
+        self.assertEqual(len(plus_tags_after), 1, "Unlinking a report line whose tags are used on move lines should keep the right sign.")
+        self.assertEqual(plus_tags_after.active, False, "Unlinking a report line whose tags are used on move lines should archive the right sign.")
+        self.assertEqual(len(test_tax.invoice_repartition_line_ids.tag_ids), 0, "After a tag is archived it shouldn't be on tax repartition lines.")
+
+    def test_unlink_report_line_tags_used_by_other_expression(self):
+        """
+        Deletion of a report line whose tags are still referenced in other expression should not delete nor archive tags.
+        """
+        tag_name = self.tax_report_line_1_1.expression_ids.formula  # tag "O1" is used in both line 1.1 and line 2.1
+        tags_before = self._get_tax_tags(self.test_country_1, tag_name=tag_name, active_test=False)
+        tags_archived_before = tags_before.filtered(lambda tag: not tag.active)
+        self.tax_report_line_1_1.unlink()
+        tags_after = self._get_tax_tags(self.test_country_1, tag_name=tag_name, active_test=False)
+        tags_archived_after = tags_after.filtered(lambda tag: not tag.active)
+        self.assertEqual(len(tags_after), len(tags_before), "Unlinking a report line whose tags are used on other expression should not delete them.")
+        self.assertEqual(len(tags_archived_after), len(tags_archived_before), "Unlinking a report line whose tags are used on other expression should not archive them.")
+
+    def test_tag_recreation_archived(self):
```

Comment:
A small docstring explaining this case would be welcome (it's a bit more advanced, and worth making sure anybody coming here understands it)